contracts/axone-objectarium/src/state.rs

```diff
@@ -194,14 +194,58 @@ impl From<msg::BucketLimits> for BucketLimits {
     }
 }
 
-impl From<BucketLimits> for msg::BucketLimits {
-    fn from(limits: BucketLimits) -> Self {
-        msg::BucketLimits {
-            max_total_size: limits.max_total_size,
-            max_objects: limits.max_objects,
-            max_object_size: limits.max_object_size,
-            max_object_pins: limits.max_object_pins,
+impl BucketLimits {
+    fn try_new(
+        max_total_size: Option<Uint128>,
+        max_objects: Option<Uint128>,
+        max_object_size: Option<Uint128>,
+        max_object_pins: Option<Uint128>,
+    ) -> StdResult<BucketLimits> {
+        if let Some(max_total_size) = max_total_size {
+            if max_total_size == Uint128::zero() {
+                return Err(StdError::generic_err("'max_total_size' cannot be zero"));
+            }
```

Comment:
I always feel like using the [`ensure!`](https://docs.rs/cosmwasm-std/latest/cosmwasm_std/macro.ensure.html) macro in such cases as it conveys the intent of a precondition that must be met.

In the specific case of checking if the `max_total_size` is not 0,  we can streamline the precondition check to a single line using the  [`ensure_ne`](https://docs.rs/cosmwasm-std/latest/cosmwasm_std/macro.ensure_ne.html) macro:

```rust
ensure_ne!(max_total_size, Some(Uint128::zero()), StdError::generic_err("'max_total_size' cannot be zero"));
```

Reply:
Oh yes you're right, I have a short memory 😅